src/Service/UrlDecodeUrlDecorator.php

```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace App\Service;
+
+class UrlDecodeUrlDecorator extends BaseUrlDecorator
```

Comment:
This is not a decorator pattern. Instead of inheritance you should implement the same interface and pass the decorated class to the constructor, to call it whenever the fallback to the original logic is needed